pandas/core/arrays/datetimes.py

```diff
@@ -2215,6 +2218,10 @@ def objects_to_datetime64ns(
             return values.view("i8"), tz_parsed
         except (ValueError, TypeError):
             raise err
+    except OverflowError as err:
+        # Exception is raised when a part of date is greater than 32 bit signed int
+        # TODO: Remove this when its fixed in Python
+        raise OutOfBoundsDatetime("datetime out of range") from err
```

Comment:
this should be more informative, like mirroring other messages like this e.g. 
```raise OutOfBoundsDatetime(f"Out of bounds nanosecond timestamp {ts}")```

Reply:
From where will I get `ts`? Its an array and the `err` variable has no information about the timestamp which caused the error.

Reply:
no that was an example. use the message